stl/inc/format

```diff
@@ -2859,122 +2862,143 @@ _NODISCARD auto make_wformat_args(const _Args&... _Vals) {
     return _Format_arg_store<wformat_context, _Args...>{_Vals...};
 }
 
-template <class _Out, class _CharT>
-using format_args_t = basic_format_args<basic_format_context<_Out, _CharT>>;
+template <class _CharT, class... _Args>
+struct _Basic_format_string {
+    basic_string_view<_CharT> _Str;
+
+    // TRANSITION, consteval compiler support
+    template <class _Ty, enable_if_t<convertible_to<const _Ty&, basic_string_view<_CharT>>, int> = 0>
+    constexpr _Basic_format_string(const _Ty& _Str_) : _Str{_Str_} {}
+};
+
+template <class... _Args>
+using _Fmt_string = _Basic_format_string<char, type_identity_t<_Args>...>;
+
+template <class... _Args>
+using _Fmt_wstring = _Basic_format_string<wchar_t, type_identity_t<_Args>...>;

```

Comment:
is the `type_identity_t` actually needed? It's OK to keep it if not (it might save the compiler some work).

Reply:
I also wondered about this (but didn't ask) - on Discord, @timsong-cpp explained that it prevents function template argument deduction from considering the first argument of `format(const _Fmt_wstring<_Types...> _Fmt, const _Types&... _Args)`, so only `const _Types&... _Args` will be used for deduction. This usage of alias templates is relatively rare in the STL, which is why we were surprised, but nothing's wrong with it and it enhances clarity, so it's good to keep.